modules/tinymce/src/themes/silver/main/ts/ui/menus/item/structure/ItemStructure.ts

```diff
@@ -83,11 +83,12 @@ const renderColorStructure = (item: ItemStructureSpec, providerBackstage: UiFact
 };
 
 const renderItemDomStructure = (ariaLabel: Optional<string>): RawDomSchema => {
-  const domTitle = ariaLabel.map((label): { attributes?: { title: string }} => ({
+  const domTitle = ariaLabel.map((label): { attributes?: { title: string; id?: string }} => ({
     attributes: {
       // TODO: AP-213 change this temporary solution to use tooltips, ensure its aria readable still.
       // for icon only implementations we need either a title or aria label to satisfy aria requirements.
-      title: I18n.translate(label)
+      title: I18n.translate(label),
+      id: Id.generate('menu-item')
```

Comment:
```suggestion
      id: Id.generate(`${id}-menu-item`)
```
Not sure if this is what you intend, but since you're pulling in an id attribute.

Reply:
Could maybe also be
```suggestion
      id: Type.isNonNullable(id) ? id : Id.generate('menu-item')
```

Agree with Mitch that the current code doesn't seem right.

Reply:
`Id.generate(prefix)` returns a string in the format `{prefix}_{id}` where `id` is auto generated by the function. An example of how this function is used is in [AriaControl.ts](https://github.com/tinymce/tinymce/blob/2d8c1c034e8aa484b868a0c44605489ee0ca9cd4/modules/alloy/src/main/ts/ephox/alloy/aria/AriaControls.ts#L30)